cirq/contrib/quirk/cells/measurement_cells.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Iterator, Optional, cast, Iterable, TYPE_CHECKING
+
+from cirq import ops
+from cirq.contrib.quirk.cells.cell import CellMaker, ExplicitOperationsCell
+
+if TYPE_CHECKING:
+    import cirq
+
+
+def generate_all_measurement_cell_makers() -> Iterator[CellMaker]:
+    yield reg_measurement("Measure")
+    yield reg_measurement("ZDetector")
+    yield reg_measurement("YDetector", basis_change=ops.X**-0.5)
+    yield reg_measurement("XDetector", basis_change=ops.Y**0.5)
+
+
+def reg_measurement(identifier: str,
+                    basis_change: Optional['cirq.Gate'] = None) -> CellMaker:
+    return CellMaker(
+        identifier=identifier,
+        size=1,
+        maker=lambda args: ExplicitOperationsCell(
+            [ops.measure(*args.qubits, key=f'row={args.row},col={args.col}')],
+            basis_change=cast(Iterable['cirq.Operation'], [
```

Comment:
Can we get rid of the cast? Gate.on() does have return type annotation, so it must be something else.

Reply:
The problem is actually with the "x if y else z". mypy infers a too-specific type from x, and then complains about z.

Reply:
The issue is the ternary expression. Mypy infers a too-specific type from the true-case value.